test/helpers.lua

```diff
@@ -741,17 +741,46 @@ function module.read_file_list(filename, start)
   if not file then
     return nil
   end
+
+  local chars_per_line = 256
+  -- `wc` indicates that there is on average 128 chars per line - use twice
+  -- that to ensure we will return as many lines as requested
+  local max_chars = chars_per_line * maxlines
+  local file_size = file:seek("end")
+  local starting_offset = file_size - max_chars
+
+  file:seek("set", starting_offset)
+
   local lines = {}
   local i = 1
-  for line in file:lines() do
+  local line = file:read("*l")
+  while line ~= nil do
```

Comment:
I think it's only reasonable to try this out `if maxlines`

Code sketch (I did not actually run it)

```lua
if maxlines and maxlines ~= 0 then
  while true do
    local line = file:read("*l")
    if line == nil then
      break
    end
    table.insert(lines, line)
    if #lines == maxlines then
      break
    end
  end
end
```